src/events/events.ts

```diff
@@ -310,31 +221,115 @@ export const onMessageEdit = async (client: Client, before: Message, after: Mess
+    if (!embed) {
         return;
     }
 
-    const embed: EmbedBuilder = genMessageDeleteEmbed(message);
+    await logTo(client, server, embed);
+};
 
-    // If embed is null, the message wasn't significant to Sushi Bot
-    if (!embed) {
+/**
+ * Broadcasts auto go-live posts and auto shout outs to sevrers subscribed to them.
+ * 
+ * @param client the Discord bot
+ * @param before the presence before update
+ * @param after the presence after update
+ */
+export const onPresenceUpdate = async (client: Client, before: Presence, after: Presence): Promise<void> => {
+    if (
+        !before || !after ||
+        before.status === "offline" ||
+        after.status === "offline"
+    ) {
         return;
     }
 
-    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+    const guild: Guild = after.guild;
+    const server: Server = await Server.get(guild.id);
+    const member: GuildMember = after.member;
+    const oldStreams: Activity[] = [];
+    let streams: Activity[] = [];
+
+    // Extract Twitch stream activites
+    for (const activity of before.activities) {
+        if (
+            activity.type === ActivityType.Streaming &&
+            activity.name === "Twitch"
+        ) {
+            oldStreams.push(activity);
+            break;
+        }
+    }
 
-    try {
-        // Log to the logs channel
-        await logs.send({ embeds: [embed] });
-    } catch {
-        // Remove configurations if logging failed.
-        // Most likely causes are that the channel was deleted or that
-        // Sushi Bot had its permissions revoked
-        server.logs = null;
-        await server.save();
+    for (const activity of after.activities) {
+        if (
+            activity.type === ActivityType.Streaming &&
+            activity.name === "Twitch"
+        ) {
+            streams.push(activity);
+            break;
+        }
+    }
+
+    // Skip if no stream was found
+    if (!streams.length) {
+        return;
+    }
+    
+    // Check if the stream link has been updated
+    if (oldStreams.length) {
+        const oldUrls: string[] = oldStreams.map(o => o.url);
+        streams = streams.filter(s => !oldUrls.includes(s.url));
+    }
+
+    // Auto go-live for server owners
+    // Auto shout out for non server owners
+    if (member.id === guild.ownerId) {
+        // Skip if no auto go-live configured
+        if (!server.goLive) {
+            return;
+        }
+
+        const channel: TextChannel = <TextChannel> client.channels.cache.get(server.goLive.channel);
+
+        for (const stream of streams) {
+            try {
+                await channel.send(formatGoLivePost(stream, server.goLive.message));
+            } catch (e) {
+                // Remove auto go-live configuration if we fail to send
+                server.goLive = null;
+                await server.save();
+                throw e;
+            }
+        }
+    } else {
```

Comment:
If it's a server owner going live and their server has auto go-live posts configured, make a go-live post.